File: src/main/java/com/regionrental/commands/VerifyCommand.java
package com.regionrental.commands;

import com.regionrental.RegionRental;
import org.bukkit.ChatColor;
import org.bukkit.command.Command;
import org.bukkit.command.CommandExecutor;
import org.bukkit.command.CommandSender;

import java.util.List;
import java.util.Map;

public class VerifyCommand implements CommandExecutor {

    private final RegionRental plugin;

    public VerifyCommand(RegionRental plugin) {
        this.plugin = plugin;
    }

    @Override
    @SuppressWarnings("unchecked")
    public boolean onCommand(CommandSender sender, Command command, String label, String[] args) {
        // Check permission
        if (!sender.hasPermission("regionrental.admin.verify")) {
            sender.sendMessage(plugin.getConfigManager().getMessage("no-permission"));
            return true;
        }

        // Check if verify command is enabled
        if (!plugin.getConfigManager().isEnableVerifyCommand()) {
            sender.sendMessage(ChatColor.RED + "The verify command is disabled in config.yml");
            sender.sendMessage(ChatColor.GRAY + "Enable it with: regions-config.enable-verify-command: true");
            return true;
        }

        sender.sendMessage(ChatColor.GOLD + "" + ChatColor.BOLD + "=== Region Configuration Verification ===");
        sender.sendMessage("");

        // Get verification report
        Map<String, Object> report = plugin.getRegionsConfig().getVerificationReport();

        int totalSigns = (int) report.get("totalSigns");
        int totalConfigs = (int) report.get("totalConfigs");
        List<String> missingConfigs = (List<String>) report.get("missingConfigs");
        List<String> orphanedConfigs = (List<String>) report.get("orphanedConfigs");

        int usingDefaults = missingConfigs.size();
        int withCustomSettings = totalSigns - usingDefaults;

        // Display summary
        sender.sendMessage(ChatColor.AQUA + "Summary:");
        sender.sendMessage(ChatColor.GRAY + "  Total regions with signs: " + ChatColor.WHITE + totalSigns);
        sender.sendMessage(ChatColor.GRAY + "  Using custom overrides:   " + ChatColor.WHITE + withCustomSettings);
        sender.sendMessage(ChatColor.GRAY + "  Using default settings:   " + ChatColor.WHITE + usingDefaults);
        sender.sendMessage("");

        // Display regions using defaults
        if (!missingConfigs.isEmpty()) {
            sender.sendMessage(ChatColor.GREEN + "Regions Using Defaults (" + missingConfigs.size() + "):");
            sender.sendMessage(ChatColor.GRAY + "  These regions use default values from config.yml:");
            for (String region : missingConfigs) {
                sender.sendMessage(ChatColor.GRAY + "    - " + ChatColor.WHITE + region);
            }
            sender.sendMessage(ChatColor.YELLOW + "  Use /rroverride to set custom values for these regions");
            sender.sendMessage("");
        } else {
            sender.sendMessage(ChatColor.YELLOW + "All regions with signs have custom overrides configured!");
            sender.sendMessage("");
        }

        // Display orphaned configs (potential issue)
        if (!orphanedConfigs.isEmpty()) {
            sender.sendMessage(ChatColor.GOLD + "Orphaned Configurations (" + orphanedConfigs.size() + "):");
            sender.sendMessage(ChatColor.GRAY + "  These custom configs exist but have no rental sign:");
            for (String region : orphanedConfigs) {
                sender.sendMessage(ChatColor.GRAY + "    - " + ChatColor.WHITE + region);
            }
            sender.sendMessage(ChatColor.YELLOW + "  Use /rroverride remove <region> to clean up unused configs");
            sender.sendMessage("");
        }

        // Summary message
        if (orphanedConfigs.isEmpty()) {
            sender.sendMessage(ChatColor.GREEN + "✓ No issues found - all configs are linked to signs");
        } else {
            sender.sendMessage(ChatColor.YELLOW + "⚠ Found " + orphanedConfigs.size() + " orphaned config(s) - consider cleanup");
        }

        sender.sendMessage("");
        sender.sendMessage(ChatColor.GOLD + "" + ChatColor.BOLD + "=================================");

        return true;
    }
}

File: src/main/kotlin/com/zonerental/commands/VerifyCommand.kt
package com.zonerental.commands

import com.zonerental.ZoneRental
import com.zonerental.extensions.sendMiniMessage
import org.bukkit.command.Command
import org.bukkit.command.CommandExecutor
import org.bukkit.command.CommandSender

/**
 * Command to verify region configuration integrity.
 * Shows regions using defaults and orphaned configurations.
 */
class VerifyCommand(private val plugin: ZoneRental) : CommandExecutor {

    override fun onCommand(sender: CommandSender, command: Command, label: String, args: Array<String>): Boolean {
        if (!sender.hasPermission("zonerental.admin.verify")) {
            sender.sendMessage(plugin.configManager.getMessage("no-permission"))
            return true
        }

        if (!plugin.configManager.isEnableVerifyCommand) {
            sender.sendMiniMessage("<red>The verify command is disabled in config.yml")
            sender.sendMiniMessage("<gray>Enable it with: regions-config.enable-verify-command: true")
            return true
        }

        sender.sendMiniMessage("<gold><bold>=== Region Configuration Verification ===")
        sender.sendMiniMessage("")

        // Get verification report with safe casts
        val report = plugin.regionsConfig.getVerificationReport()
        val totalSigns = report["totalSigns"] as? Int ?: 0
        val missingConfigs = (report["missingConfigs"] as? List<*>)?.filterIsInstance<String>() ?: emptyList()
        val orphanedConfigs = (report["orphanedConfigs"] as? List<*>)?.filterIsInstance<String>() ?: emptyList()

        val usingDefaults = missingConfigs.size
        val withCustomSettings = totalSigns - usingDefaults

        // Display summary
        sender.sendMiniMessage("<aqua>Summary:")
        sender.sendMiniMessage("<gray>  Total regions with signs: <white>$totalSigns")
        sender.sendMiniMessage("<gray>  Using custom overrides:   <white>$withCustomSettings")
        sender.sendMiniMessage("<gray>  Using default settings:   <white>$usingDefaults")
        sender.sendMiniMessage("")

        // Display regions using defaults
        if (missingConfigs.isNotEmpty()) {
            sender.sendMiniMessage("<green>Regions Using Defaults (${missingConfigs.size}):")
            sender.sendMiniMessage("<gray>  These regions use default values from config.yml:")
            missingConfigs.forEach { region ->
                sender.sendMiniMessage("<gray>    - <white>$region")
            }
            sender.sendMiniMessage("<yellow>  Use /zroverride to set custom values for these regions")
            sender.sendMiniMessage("")
        } else {
            sender.sendMiniMessage("<yellow>All regions with signs have custom overrides configured!")
            sender.sendMiniMessage("")
        }

        // Display orphaned configs (potential issue)
        if (orphanedConfigs.isNotEmpty()) {
            sender.sendMiniMessage("<gold>Orphaned Configurations (${orphanedConfigs.size}):")
            sender.sendMiniMessage("<gray>  These custom configs exist but have no rental sign:")
            orphanedConfigs.forEach { region ->
                sender.sendMiniMessage("<gray>    - <white>$region")
            }
            sender.sendMiniMessage("<yellow>  Use /zroverride remove <region> to clean up unused configs")
            sender.sendMiniMessage("")
        }

        // Summary message
        if (orphanedConfigs.isEmpty()) {
            sender.sendMiniMessage("<green>✓ No issues found - all configs are linked to signs")
        } else {
            sender.sendMiniMessage("<yellow>⚠ Found ${orphanedConfigs.size} orphaned config(s) - consider cleanup")
        }

        sender.sendMiniMessage("")
        sender.sendMiniMessage("<gold><bold>=================================")

        return true
    }
}
